src/frontend/src/tests/mocks/page.store.mock.ts

```diff
@@ -1,13 +1,26 @@
+import { resetRouteParams, type RouteParams } from '$lib/utils/nav.utils';
+import type { Page } from '@sveltejs/kit';
 import { writable } from 'svelte/store';
 
-export const mockPageStore = () => {
-	vi.mock('$app/stores', () => ({
-		page: writable({
-			params: {},
-			route: {},
-			status: 200,
-			error: null,
-			data: {}
-		})
-	}));
+const initialStoreValue = {
+	data: resetRouteParams(),
+	route: {
+		id: null
+	}
 };
+
+const initPageStoreMock = () => {
+	const { subscribe, set } = writable<Partial<Page>>(initialStoreValue);
+
+	return {
+		subscribe,
+		mock: (data: Partial<RouteParams>) =>
+			set({
+				data
+			}),
+
+		reset: () => set(initialStoreValue)
+	};
+};
+
+export const page = initPageStoreMock();
```

Comment:
should we name it `mockPage`?

Reply:
yes, sure! I thought the same at first, but wanted to exactly replicate NNS dapp eheh